structs.go

```diff
@@ -84,6 +89,8 @@ func (s *Struct) Map() map[string]interface{} {
 	return out
 }
 
+type KeyFunc func(reflect.Type, reflect.StructField) string
```

Comment:
I think this should be named as `TagFunc`. Also it needs a doc comment. 

Second I think we should only pass `reflect.StructField`. It's for naming and I don't want to pass any `reflect.Type` or `reflect.Value` arguments to create any kind of side effects.